addons/stock/models/stock_quant.py

```diff
@@ -176,6 +186,42 @@ def _update_dict(date_by_quant, key, value):
         for quant in self:
             quant.last_count_date = date_by_quant.get((quant.location_id.id, quant.package_id.id, quant.product_id.id, quant.lot_id.id, quant.owner_id.id))
 
+    def _search_last_count_date(self, operator, value):
+        op = PY_OPERATORS.get(operator)
+        if not op:
+            return NotImplemented
+        domain = [('state', '=', 'done'), ('is_inventory', '=', True)]
+        if value is not False:
+            if operator == '=':
+                domain += [('date', '>=', value), ('date', '<=', value)]
+            else:
+                domain.append(('date', operator, value))
+        groups = self.env['stock.move.line']._read_group(
+            domain=domain,
+            aggregates=['id:recordset'],
+        )
+        move_lines = groups[0][0]
+        quant_domain = Domain.AND([
+            Domain('product_id', 'in', move_lines.product_id.ids),
+            Domain.OR([
+                Domain('lot_id', 'in', move_lines.lot_id.ids),
+                Domain('lot_id', '=', False),
+            ]),
+            Domain.OR([
+                Domain('owner_id', 'in', move_lines.owner_id.ids),
+                Domain('owner_id', '=', False),
+            ]),
```

Comment:
```suggestion
        quant_domain = Domain.AND([
            Domain('product_id', 'in', move_lines.product_id.ids),
            Domain('lot_id', 'in', move_lines.lot_id.ids + [False]),
            Domain('owner_id', 'in', move_lines.owner_id.ids + [False]),
```

Please apply the `same logic for location_id and package_id`.

Right now, the domain is wrong: in the compute method, the intention is `“either location_id or location_dest_id should match the move line locations (self.location_id.ids)”`.
But your domain enforces that `location_id` must be in the `union of source or dest locations`, which has different semantics and can produce incorrect matches.